win_loss.py

```diff
@@ -0,0 +1,161 @@
+        fighter_light_heavyweight = df[df['division'] == 'Light Heavyweight']
+        fighter_heavyweight = df[df['division'] == 'Heavyweight']
+        fighter_catchweight = df[df['division'] == 'Catchweight']
+        fighter_overall = df
+        return fighter_flyweight, fighter_bantamweight, fighter_featherweight, fighter_lightweight, fighter_welterweight, fighter_middleweight, fighter_light_heavyweight, fighter_heavyweight, fighter_catchweight, fighter_overall
+
+    """
+    Filters the data by outcome method
+    """
+    def get_fighter_outcome_methods(self, df):
+        fighter_ko = df[(df['outcome_method'] == 'KO/TKO') | (df['outcome_method'] == 'TKO - Doctor\'s Stoppage')]
+        fighter_sub = df[df['outcome_method'] == 'Submission']
+        fighter_decision = df[(df['outcome_method'] == 'Decision - Unanimous') | (df['outcome_method'] == 'Decision - Split') | (df['outcome_method'] == 'Decision - Majority')]
+        fighter_total = df
+        return fighter_ko, fighter_sub, fighter_decision, fighter_total
+
+    """
+    Filters the data by outcome
+    """
+    def get_fighter_outcomes(self, df, fighter_id):
+        fighter_wins = df[df['winner_id'] == fighter_id]
+        fighter_losses = df[df['winner_id'] != fighter_id]
+        return fighter_wins, fighter_losses
+
+    """
+    Filters the data by a specific fighter
+    """
+    def get_fighter_fights(self, df, fighter_id, index):
+        year_ago = self.compute_year_ago(df)
+        all_prev_fights = df.loc[:index-1]
+        if not all_prev_fights.empty:
+            fighter_a_id_vals = all_prev_fights.fighter_a_id.values
+            fighter_b_id_vals = all_prev_fights.fighter_b_id.values
+
+            prev_fights = all_prev_fights[((fighter_a_id_vals == fighter_id) | (fighter_b_id_vals == fighter_id))]
+            prev_fights_last_year = all_prev_fights[((fighter_a_id_vals == fighter_id) | (fighter_b_id_vals == fighter_id)) & (pd.to_datetime(all_prev_fights['date']) > year_ago[index])]
+
+            return prev_fights, prev_fights_last_year
+        return pd.DataFrame(), pd.DataFrame()
+    
+    """
+    Creates the column names for the win/loss features
+    """
+    def create_col_names(self):
+        col_names = []
+        fighters = ['fighter-a', 'fighter-b']
+        outcomes = ['wins', 'losses']
+        methods = ['KO/TKO', 'submission', 'decision', 'total']
+        weight_classes = ['flyweight', 'bantamweight', 'featherweight', 'lightweight', 'welterweight', 'middleweight', 'light-heavyweight', 'heavyweight', 'catchweight', 'overall']
+        time_periods = ['last-year', 'all-time']
+
+        for fighter in fighters:
+            for outcome in outcomes:
+                for method in methods:
+                    for weight_class in weight_classes:
+                        for time_period in time_periods:
+                            col_name = f"{fighter.replace(' ', '_')}_{outcome}_{method}_{weight_class.replace(' ', '_')}_{time_period}"
```

Comment:
.replace('-', '_') was it meant to be this? and why not replace the space in time_period string?